### PR TITLE
Added some federation tweaks

### PR DESCRIPTION
**Pull request by alansikora**
- Made sure development environment does not use production federation hub;
 - Organized some files;
 - Added federation statistics;

### CODE REVIEW

File: app/statistics/server/functions/get.js [hunk 1]
```diff
@@ -71,10 +77,16 @@ statistics.get = function _getStatistics() {
...
+	statistics.onlineUsers = Meteor.users
+		.find({ statusConnection: 'online' })
+		.count();
+	statistics.awayUsers = Meteor.users
+		.find({ statusConnection: 'away' })
+		.count();
+	statistics.totalConnectedUsers =
+		statistics.onlineUsers + statistics.awayUsers;
+	statistics.offlineUsers =
+		statistics.totalUsers - statistics.onlineUsers - statistics.awayUsers;
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]: are all of the second line things something your IDE is doing?

File: app/statistics/server/functions/get.js [hunk 2]
```diff
@@ -129,16 +175,57 @@ statistics.get = function _getStatistics() {
...
+		})
+		.count();
+
+	const { mongo } = MongoInternals.defaultRemoteCollectionDriver();
 
-	const { oplogEnabled, mongoVersion, mongoStorageEngine } = getMongoInfo();
-	statistics.oplogEnabled = oplogEnabled;
-	statistics.mongoVersion = mongoVersion;
-	statistics.mongoStorageEngine = mongoStorageEngine;
+	if (mongo._oplogHandle && mongo._oplogHandle.onOplogEntry) {
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]: These changes here we'll definitely want whats in develop or will re-break some some issues

Comment (alansikora) [reply app/statistics/server/functions/get.js]: That was not supposed to be here, act like it never happened 😆

File: app/statistics/server/functions/get.js (hunk 1 above)
Comment (alansikora) [reply app/statistics/server/functions/get.js]: Yeah, I had an issue with my VSCode, I'm pushing a new version

**Comment by alansikora** [comment]
@geekgonecrazy you can check again

File: app/statistics/server/functions/get.js [hunk 3]
```diff
@@ -134,11 +144,48 @@ statistics.get = function _getStatistics() {
 	statistics.migration = Migrations._getControl();
 	statistics.instanceCount = InstanceStatus.getCollection().find({ _updatedAt: { $gt: new Date(Date.now() - process.uptime() * 1000 - 2000) } }).count();
 
+	const { mongo } = MongoInternals.defaultRemoteCollectionDriver();
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]:
```suggestion
```

File: app/statistics/server/functions/get.js [hunk 4]
```diff
@@ -134,11 +144,48 @@ statistics.get = function _getStatistics() {
...
+
 	const { oplogEnabled, mongoVersion, mongoStorageEngine } = getMongoInfo();
 	statistics.oplogEnabled = oplogEnabled;
 	statistics.mongoVersion = mongoVersion;
 	statistics.mongoStorageEngine = mongoStorageEngine;
 
+	if (mongo._oplogHandle && mongo._oplogHandle.onOplogEntry) {
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]:
```suggestion
```

File: app/statistics/server/functions/get.js [hunk 5]
```diff
@@ -134,11 +144,48 @@ statistics.get = function _getStatistics() {
...
+		statistics.oplogEnabled = true;
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]:
```suggestion
```

File: app/statistics/server/functions/get.js [hunk 6]
```diff
@@ -134,11 +144,48 @@ statistics.get = function _getStatistics() {
...
+	}
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]:
```suggestion
```

File: app/statistics/server/functions/get.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ import os from 'os';
 
 import _ from 'underscore';
 import { Meteor } from 'meteor/meteor';
+import { MongoInternals } from 'meteor/mongo';
```
Comment (geekgonecrazy) [inline app/statistics/server/functions/get.js]:
```suggestion
```

**Comment by geekgonecrazy** [comment]
@sampaiodiego what do you want to do about these linting errors?  It is griping about:
```
20:1  error  There should be no empty line within import group  import/order`
```

Seems to only happen if a single import on its own line.  If there are two in a row it stops

**Comment by alansikora** [comment]
@geekgonecrazy that does not happen locally

**Comment by sampaiodiego** [comment]
@geekgonecrazy [there are few import groups](https://github.com/RocketChat/eslint-config-rocketchat/blob/master/rules/imports.js#L157) that cannot have empty lines between those imports.

Usually they'll be external and internal imports, they need to separate by an empty line, but cannot have empty lines within the group, like:

```javascript
import _ from 'underscore';
import { Meteor } from 'meteor/meteor';

import { anything } from './internal/path';
import { other } from './internal/other';
```

File: app/federation/server/index.js [hunk 8]
```diff
@@ -1,21 +1,21 @@
 import { Meteor } from 'meteor/meteor';
 import { _ } from 'meteor/underscore';
 
-import './federation-settings';
+import { FederationKeys } from '../../models';
+import { settings } from '../../settings';
+import { getConfig } from './config';
+import './adminSettings';
+
```
Comment (sampaiodiego) [inline app/federation/server/index.js]: this empty line needs to be removed to pass the linter =)

```suggestion
```

File: app/statistics/server/functions/get.js [hunk 9]
```diff
@@ -19,6 +19,8 @@ import { Info, getMongoInfo } from '../../../utils/server';
 import { Migrations } from '../../../migrations/server';
 import { statistics } from '../statisticsNamespace';
```
Comment (sampaiodiego) [inline app/statistics/server/functions/get.js]:
```suggestion
```

this one as well

**Comment by geekgonecrazy** [comment]
Is this coming in 1.1.0 or being delayed to 1.2.0?

**Comment by sampaiodiego** [comment]
delayed to 1.2.0

**Comment by alansikora** [comment]
@sampaiodiego I will close this one, you can reopen if you wish, but this is not relevant anymore.

**Pull request closed without merging**